frame/beefy-mmr/src/lib.rs

```diff
@@ -138,12 +120,11 @@ pub mod pallet {
 		/// efficiency reasons.
 		type BeefyAuthorityToMerkleLeaf: Convert<<Self as pallet_beefy::Config>::BeefyId, Vec<u8>>;
 
-		/// Retrieve a list of current parachain heads.
-		///
-		/// The trait is implemented for `paras` module, but since not all chains might have
-		/// parachains, and we want to keep the MMR leaf structure uniform, it's possible to use
-		/// `()` as well to simply put dummy data to the leaf.
-		type ParachainHeads: ParachainHeadsProvider;
+		/// The type expected for the leaf extra data
+		type LeafExtra: AsRef<[u8]> + Member + codec::FullCodec;
```

Comment:
Can you also please remove unnecessary `AsRef<u8>`?

```suggestion
		type LeafExtra: Member + codec::FullCodec;
```